pkg/endpoint/bpf.go

```diff
@@ -1249,14 +1319,82 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	return nil
 }
 
+// endpointPolicyLockdown puts the endpoint policy map into a lockdown
+// mode. If the policy map can still accomodate all the deny entries (and
+// there is at least one deny policy) then all deny entries are inserted without
+// any allows. If there is no space for the deny entries or there is no deny entries
+// to insert then a new bpf policy map is populated with deny all traffic entries.
+// The old map is unpinned, and the new map is pinned.
+func (e *Endpoint) endpointPolicyLockdown() (bool, error) {
+	var fullLockDown bool
+	tmpPMPath := e.policyMapPath() + "_lockdown"
+	pM, err := policymap.CreateUnpinned(tmpPMPath)
+	if err != nil {
+		return false, fmt.Errorf("could not create bpf map %q: %w", tmpPMPath, err)
+	}
+	mapState := policy.NewMapState(nil)
+	// deny entries also exceed max entries, or there are no deny entires,
+	// go into a full lockdown.
+	if e.desiredPolicy.GetPolicyMap().DenyLen() > policymap.MaxEntries ||
+		e.desiredPolicy.GetPolicyMap().DenyLen() == 0 {
+		fullLockDown = true
+		for _, k := range allTrafficKeys {
+			policymapKey := policymap.NewKey(k.Identity, k.DestPort,
+				k.Nexthdr, k.TrafficDirection)
+			if err := pM.DenyKey(policymapKey); err != nil {
+				return false, fmt.Errorf("failed to add deny all policy: %w", err)
+			}
+			mapState.Insert(k, policy.MapStateEntry{IsDeny: true})
+		}
+	} else {
+		var outErr error
+		e.desiredPolicy.GetPolicyMap().ForEachDeny(func(k policy.Key, entry policy.MapStateEntry) bool {
```

Comment:
If I follow this correctly we are inserting denies first, when are the allows? Maybe the misleading part is the `ForEachDeny` method name.

Reply:
Sorry. I made a mistake. There will never be allows.